includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -1365,6 +1367,8 @@ public function update_existing_intent( $intent, $order, $prepared_source ) {
 			$request['customer'] = $prepared_source->customer;
 		}
 
+		$request['payment_method_types'] = [ 'card' ];
```

Comment:
Also, why is this needed? I'm afraid it may change current behavior.

Reply:
This function is only being used by stripe credit card payment. Maybe it should be inside `WC_Gateway_Stripe` instead of `WC_Stripe_Payment_Gateway`.

This change is necessary in case a customer tries to pay with Boleto or OXXO but it fails,  then the customer tries to checkout with credit card.